### ISSUE TITLE
Upgrade @nuxtjs/supabase from 0.3.x to 1.0.x: 500 error > invalid claim: missing sub claim

### ISSUE THREAD

**Issue by mendrinos** [nuxt-modules/supabase#238]
### Version
@nuxtjs/supabase: 1.0.2
nuxt: 3.6.5

### Steps to reproduce
Use Nuxt 3.6.5 with @nuxtjs/supabase: 0.3.7
Then upgrade to @nuxtjs/supabase: 1.0.2
Follow the breaking changes and update your code as per the guidance: 
https://supabase.nuxtjs.org/changelog

### What is Expected?

To see authentication functioning without any defects, as I followed the upgrade steps, especially the ones with the "breaking changes" across my codebase.

### What is actually happening?

Getting error 500 across all routes
invalid claim: missing sub claim

at createError (./node_modules/h3/dist/index.mjs:127:15)
at serverSupabaseUser (./.nuxt/dev/index.mjs:776:11)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async Object.handler (./.nuxt/dev/index.mjs:783:16)
at async Object.handler (./node_modules/h3/dist/index.mjs:1284:19)
at async toNodeHandle (./node_modules/h3/dist/index.mjs:1359:7)
at async Object.ufetch [as localFetch] (./node_modules/unenv/runtime/fetch/index.mjs:9:17)
at async Object.errorhandler [as onError] (./.nuxt/dev/index.mjs:665:30)
at async Server.toNodeHandle (./node_modules/h3/dist/index.mjs:1366:9)

Under Auth Logs from supabase admin I see 5 times this log:

Event Message
{"component":"api","error":"401: invalid claim: missing sub claim","level":"info","method":"GET","msg":"401: invalid claim: missing sub claim","path":"/user","referer":"","remote_addr":"x.x.x.x","time":"2023-08-07T11:45:29Z","timestamp":"2023-08-07T11:45:29Z"}

I am suspecting it has to do with the switch to the new default authentication method of PCKE.

**Comment by bbaltatu** [comment]
I succeed to reproduce by using the official supabase URL and KEY (not local supabase) and inside a docker container
My docker file

`# Folosim o imagine node versiunea 18 pentru construirea aplicației
FROM node:18.17-bullseye as build-stage

WORKDIR /app

# Copiem fișierele `package.json` și `package-lock.json` din rădăcina monorepo-ului, dacă există
COPY package*.json ./

# Copiem fișierele de configurare ale aplicației și ale pachetelor din `packages/`
COPY ./apps/ro-idei/package*.json ./apps/ro-idei/
COPY ./packages/*/package*.json ./packages/

# Copiem sursele pachetelor de care depinde aplicația
COPY ./packages/ ./packages/

# Copiem codul sursă al aplicației
COPY ./apps/ro-idei ./apps/ro-idei

# Instalăm dependințele la nivelul de rădăcină folosind npm și construim aplicația
RUN npm install
WORKDIR /app/apps/ro-idei
RUN npm run build

# În stadiul de producție, creăm o nouă imagine pentru a rula aplicația
FROM node:18-alpine as production-stage

WORKDIR /app

# Copiem fișierele necesare pentru a rula aplicația din stadiul de construire
COPY --from=build-stage /app/apps/ro-idei/.output .

# Setăm portul și comanda de start pentru aplicație
EXPOSE 3000
CMD [ "node", "server/index.mjs" ]
`

**Comment by bbaltatu** [comment]
![image](https://github.com/nuxt-modules/supabase/assets/11313025/33510166-b463-46aa-b4e1-2a4742d4df29)
Still digging to issue.... Any answer?

**Comment by bbaltatu** [comment]
I suppose I know the cause https://github.com/supabase/supabase-js/issues/702

serverSupabaseUser returns that error because the user is not logged in and the token is invalid.  
I am looking for a solution. I will come back if I will found one.

**Comment by CptJJ** [comment]
Had this issue, fixed by doing this on all ym server routes that use the serverSupabaseUser:

```js
var user = null
  try {
    user = await serverSupabaseUser(event)
  } catch (err) {

  }
  ```

**Comment by mendrinos** [comment]
Thank you @CptJJ, the above fixed it on my end for the time being.

**Comment by mendrinos** [comment]
@larbish any update? seems the magic link authentication cannot function at all.

**Comment by larbish** [comment]
If this is related to the PKCE flow, did you try to use old auth method ?

**Comment by mendrinos** [comment]
Thanks @larbish ,

I switched to 'implicit' flow and it worked just fine.

`clientOptions: {
      auth: {
        flowType: 'implicit',
        detectSessionInUrl: true,
        persistSession: true,
        autoRefreshToken: true
      },
    }`

**Comment by peterkimzz** [comment]
I don't think getting the `invalid claim error` after calling `signOut` method is normal behaviour. (including it throws 500 code)

I can skip the error in server side like below,

```ts
export const getSupabaseUser = async (event: H3Event) => {
  const client = await serverSupabaseClient(event);
  const {
    data: { user },
    error,
  } = await client.auth.getUser();

  if (error) {
    console.log("getSupabaseUser error", error);
  }

  return { supabaseUser: user };
};
```

**Comment by CosAnca** [comment]
I'm facing the same issue and switching to `implicit` flow doesn't work either in my case.
The solution provided by @peterkimzz seems to prevent the application to throw 500 and on logout the user is correctly redirected to the `/login` page but the error is still fired on the server and having other links on the `/login` page (for example a NuxtLink to a `/register` page) makes them unusable.

**Comment by ammuench** [comment]
Just opened https://github.com/nuxt-modules/supabase/pull/272 to hopefully address this issue.

Currently the `useSupabaseUser` composable is a synchronous method with a promise within--changing it to be a proper async method resolved the `invalid claim: missing sub claim` when making calls to my backend that had a `serverSupabaseUser` call after login

**Comment by chiuwah** [comment]
I am on v1.1.2 and also getting error 500 (invalid claim: missing sub claim) when using serverSupabaseUser() without the user logged in on server api route. Any solutions available?

**Comment by kylegl** [comment]
I made a reproduction of where im getting this error.  I use the `serverSupabaseUser` on the backend in my `trpc` context file. `/server/trpc/context.ts`. Is this related to #272?

[nuxt-trpc-prisma-supabase-repro](https://codesandbox.io/p/github/kylegl/nuxt-trpc-prisma/main?file=%2Fpackage.json%3A1%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cln5v7bpk0007356f6z1vycjc%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cln5v7bpj0003356fiaw1qlar%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cln5v7bpj0005356fjxet2pym%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cln5v7bpk0006356f849qf8cm%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cln5v7bpj0003356fiaw1qlar%2522%253A%257B%2522id%2522%253A%2522cln5v7bpj0003356fiaw1qlar%2522%252C%2522activeTabId%2522%253A%2522clo2btzsd02df356ikow6qhzg%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cln5v7bpj0002356furcw8zuj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522cln5v7k0q006y356fsen09idd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpages%252Findex.vue%2522%252C%2522id%2522%253A%2522clo2aieg200au356i1r70r0ee%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fnuxt.config.ts%2522%252C%2522id%2522%253A%2522clo2b699b00qb356iuhvbtq4j%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fserver%252Ftrpc%252Fcontext.ts%2522%252C%2522id%2522%253A%2522clo2b711i00ti356i0cq42wd8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.vscode%252Fsettings.json%2522%252C%2522id%2522%253A%2522clo2btzsd02df356ikow6qhzg%2522%252C%2522mode%2522%253A%2522temporary%2522%257D%255D%257D%252C%2522cln5v7bpk0006356f849qf8cm%2522%253A%257B%2522id%2522%253A%2522cln5v7bpk0006356f849qf8cm%2522%252C%2522activeTabId%2522%253A%2522clo2bt7780282356iygkfqu2y%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522PROJECT_SETUP%2522%252C%2522id%2522%253A%2522clo1a03wb008m356ics48oxj7%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A46137%252C%2522id%2522%253A%2522clo2bt7780282356iygkfqu2y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522cln5v7bpj0005356fjxet2pym%2522%253A%257B%2522id%2522%253A%2522cln5v7bpj0005356fjxet2pym%2522%252C%2522activeTabId%2522%253A%2522cln5vzai9007o356f1oh13h24%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cln5vzai9007o356f1oh13h24%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clo2boown001jeffr1zu3906t%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

@larbish

**Comment by larbish** [comment]
Can someone provide a simple reproduction? I'm not able to reproduce.

**Comment by kylegl** [comment]
@larbish was my repro not working for you?

**Comment by larbish** [comment]
Sorry, missed it! I'll have a check this week.

**Comment by andreasvirkus** [comment]
@larbish have you had the time to check it out?

**Comment by larbish** [comment]
Should be fixed in version `1.1.5`. Could someone confirm ?
